modules/gapi/src/streaming/onevpl/engine/preproc_engine_interface.hpp

```diff
@@ -0,0 +1,35 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2022 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_ENGINE_PROCESSING_ENGINE_INTERFACE_HPP
+#define GAPI_STREAMING_ONEVPL_ENGINE_PROCESSING_ENGINE_INTERFACE_HPP
+
+#include "precomp.hpp"
+#include <opencv2/gapi/media.hpp>
+#include <opencv2/gapi/util/optional.hpp>
+
+#include "streaming\onevpl\engine\preproc_defines.hpp"
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+struct IPreprocEngine {
```

Comment:
This interface should be placed into onevlp namespace, or should be freed from onevpl dependencies. Is it proposed as a generic preprocessing interface or as a vpl-specific preprocessing interface?

Reply:
it's considered as generic: i believe that current  ONNX preprocessing can be wrapped by this interface to follow the single-style way

But because there are no consumers for this functionality - i've placed it in one_vpl namespace with possible undercover it for `cv::gapi` ( by demand)